Controller/Index/PaymentActionAbstract.php

```diff
@@ -0,0 +1,321 @@
+                    break;
+                case 'object':
+                    if ($this->getEventMethod() instanceof \Closure) {
+                        $eventMethodString = 'anonymous function';
+                        break;
+                    }
+                    $eventMethodString = get_class($this->getEventMethod());
+                    break;
+            }
+        }
+
+        return $eventMethodString;
+    }
+
+    /**
+     * Sets JSON result
+     *
+     * @param string $message
+     * @param int $httpCode
+     * @return JsonResult
+     */
+    protected function setResult($message = '', $httpCode = 200)
+    {
+        $result = $this->resultJsonFactory->create();
+        $result->setData(array_merge($this->getEventArgs(), ['result' => $message]));
+        $result->setHttpResponseCode($httpCode);
+
+        return $result;
+    }
+
+    /**
+     * Sets JSON response
+     *
+     * @param string $key
+     * @param string $message
+     * @param int $httpCode
+     * @return JsonResult
+     */
+    protected function setResponse($key, $message = '', $httpCode = 200)
+    {
+        $result = $this->resultJsonFactory->create();
+        $result->setData(array_merge($this->getEventArgs(), [$key => $message]));
+        $result->setHttpResponseCode($httpCode);
+
+        return $result;
+    }
+
+    protected function setRedirect($url, $code = 301)
+    {
+        /** @var \Magento\Framework\App\Response\HttpInterface $response */
+        $response = $this->getResponse();
+        $response->setRedirect($url, $code)->sendResponse();
+    }
+
+    /**
+     * @return ResponseInterface|JsonResult|ResultInterface
+     *
+     * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+     * @SuppressWarnings(PHPMD.NPathComplexity)
+     */
+    public function execute()
+    {
+        if (!$this->configHelper->isActive()) {
+            $this->logger->error(
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                basename(get_class($this)) . ' trigger error: Module is not active'
+            );
+            return $this->setResult(
+                __('Not Found: The required SwedbankPay Payments resources do not seem to be available'),
+                404
+            );
+        }
+
+        /** @noinspection PhpUndefinedMethodInspection */
+        $requestBody = json_decode($this->getRequest()->getContent());
+
+        $this->eventManager->dispatch(
+            $this->getFullEventName('before'),
+            ['requestBody' => $requestBody]
+        );
+
+        try {
+            $eventArgs = [];
+            foreach (array_keys($this->getEventArgs()) as $arg) {
+                $eventArgs[$arg] = $requestBody->{$arg};
+            }
+            $this->setEventArgs($eventArgs);
+            $this->logger->debug(
+                // phpcs:ignore Magento2.Functions.DiscouragedFunction
+                basename(get_class($this)) . ' triggered',
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The use of function basename() is discouraged](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)